src/distPlus/distribution/DistTypes.re

```diff
@@ -14,20 +14,36 @@ type xyShape = {
   ys: array(float),
 };
 
+type interpolation = [
+    | `Stepwise
+    | `Linear
+];
+type extrapolation = [
```

Comment:
Maybe a different name would be, ``assumedOutOfRangeYValues``, then the terms could be `Zero and `NearestPoint.

Reply:
Related, when would we want to use ``UseOutermostPoints``? That seems pretty dangerous to me, though I could see the appeal. I could imagine this could cause a whole lot of error.

Reply:
It's needed when combining integrals, so the CDF plot doesn't drop to zero at the right edge of the integral xyShapes.

Reply:
Ah, this makes it scarier to me now that I understand. 

I don't like that it's like one step removed from what's going on. Like, the essential thing in the codebase is that pdfs are distributions, defined as having 0s on their sides. CDFs aren't actually distributions, but are distribution functions. It's weird to have functions for distribution functions mixed in with the Continuous distributions code. 

Ideally we'd do something more like have a,
```
distributionType = [|`PDF | `CDF |]
```

Then pass this in as needed, and have lower level parts of the codebase figure out which extrapolation method to use based on that.

Reply:
Okay, done that!

As I've said I really don't love the way the whole CDF stuff is handled right now, but everything seems to be working okay at least and I think it's not excessively abstruse.